core/tests/recovery-test/tests/genesis-recovery.test.ts

```diff
@@ -0,0 +1,271 @@
+import { expect } from 'chai';
+import * as zkweb3 from 'zksync-web3';
+import fs, { FileHandle } from 'node:fs/promises';
+import { ChildProcess, spawn } from 'node:child_process';
+import path from 'node:path';
+import * as zksync from 'zksync-ethers';
+
+import {
+    externalNodeArgs,
+    getExternalNodeHealth,
+    killExternalNode,
+    NodeComponents,
+    sleep,
+    stopExternalNode,
+    waitForProcess
+} from '../src';
+
+// FIXME: check consistency checker health once it has acceptable speed
```

Comment:
I've initially added consistency checker health checks to the new test; however, they've turned out to make the test exceedingly slow (order of 3 min) because of the slow commitment generator. I've created a task to tackle this separately.